include/nlohmann/json_fwd.hpp

```diff
@@ -31,6 +31,7 @@ template<template<typename U, typename V, typename... Args> class ObjectType =
          class NumberIntegerType = std::int64_t,
          class NumberUnsignedType = std::uint64_t,
          class NumberFloatType = double,
+         class BinaryType = std::vector<std::uint8_t>,
```

Comment:
Maybe use `std::byte` when C++17? That type was added exactly for such purposes.  
Or at least consider a possible define for this.

Reply:
I’d like to keep it uint8, because this is also use in the binary readers and writers. Maybe a template type could help in the future.

Reply:
I have been avoiding using C++14 and beyond in this code base because the library is advertised as working on very old versions of GCC (like 4.8), otherwise, there's also a [[maybe_unused]] attribute flag in C++17 that I want to use as well to silence some spurious warnings about casting (but are actually unused variables.)

Reply:
For attributes, I recently started using [Hedley](https://nemequ.github.io/hedley/) which allows a portable way to use all kind of attributes.